janusgraph-core/src/main/java/org/janusgraph/graphdb/database/management/AbstractIndexStatusWatcher.java

```diff
@@ -22,34 +22,40 @@
 import java.time.Duration;
 import java.time.temporal.TemporalUnit;
 import java.util.concurrent.Callable;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
 
 public abstract class AbstractIndexStatusWatcher<R, S extends AbstractIndexStatusWatcher<R,S>> implements Callable<R> {
 
     protected JanusGraph g;
-    protected SchemaStatus status;
+    protected ArrayList<SchemaStatus> statuses = new ArrayList<SchemaStatus>();
     protected Duration timeout;
     protected Duration poll;
 
     public AbstractIndexStatusWatcher(JanusGraph g) {
         this.g = g;
-        this.status = SchemaStatus.REGISTERED;
+        this.statuses.add(SchemaStatus.REGISTERED);
         this.timeout = Duration.ofSeconds(60L);
         this.poll = Duration.ofMillis(500L);
     }
 
     protected abstract S self();
 
     /**
-     * Set the target index status.  {@link #call()} will repeatedly
+     * Set the target index statuses.  {@link #call()} will repeatedly
      * poll the graph passed into this instance during construction to
      * see whether the index (also passed in during construction) has
-     * the supplied status.
+     * one of the the supplied statuses.
      *
      * @param status
      * @return
      */
-    public S status(SchemaStatus status) {
-        this.status = status;
+    public S status(SchemaStatus... statuses) {
+        this.statuses.clear();
+        for(int i = 0; i < statuses.length; i++) {
+            this.statuses.add(statuses[i]);
```

Comment:
I changed to `this.statuses.addAll(Arrays.asList(statuses))`; ya like that better?